pkg/util/cert/crl.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cert
+
+import (
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"errors"
+	"fmt"
+	"io/ioutil"
+)
+
+// RevocationPolicy is a strategy for querying CRLs and evaluating if a certificate
+// has been revoked.
+type RevocationPolicy interface {
+	VerifyCertificate(cert *x509.Certificate) error
```

Comment:
I think returning of `(valid bool, err error)` would help in distinguishing between invalid certs and errors verifying